### PR TITLE
fix: lnurl scheme parsing

### LINKED ISSUE #2927: [BUG] non-LNURL link with "lnurl" in it incorrectly opens Alby extension
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

<a href="https://files.ekzyis.com/public/sn/lnurlpay_comments.mp4">lnurlpay_comments.mp4</a> opens the extension when it should not.

### Screenshots [optional]

_No response_

### Steps To Reproduce

1. Click the link lnurlpay_comments link in https://files.ekzyis.com/public/sn/

### Expected behavior

Alby does not intercept links that are not actually LNURLs

### Alby information

3.4.1

### Device information

_No response_

### Additional context

_No response_

### Are you working on this?

None

### PR DESCRIPTION
**Pull request by ekzyis**
### Describe the changes you have made in this PR

In #2927, I noticed that links that start with `lnurl` open the Alby extension. This means that even links to content unrelated to lightning like `lnurlpay_comments.mp4` open the Alby extension.

I looked up what schemes are meant to be parsed and I found [LUD-17](https://github.com/lnurl/luds/blob/luds/17.md).

So I think only the following link prefixes should open the Alby extension:

`lightning:`, `lnurlc:`, `lnurlw:`, `lnurlp:`

### Link this PR to an issue [optional]

Fixes #2927

### Type of change

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Screenshots of the changes [optional]

<details>

<summary>video</summary>

https://github.com/getAlby/lightning-browser-extension/assets/27162016/ad269ac5-994f-4a4a-a81c-4a91f2e104ce

</details>

### How has this been tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [ ] Added automated tests where applicable
- [ ] Update Docs & Guides
- For UI-related changes
- [ ] Darkmode
- [ ] Responsive layout

### CODE REVIEW

File: src/extension/inpage-script/index.js [hunk 1]
```diff
@@ -45,10 +45,12 @@ function registerLightningLinkClickHandler() {
...
       }
       const lightningLink = target.closest('[href^="lightning:" i]');
-      const lnurlLink = target.closest('[href^="lnurl" i]');
+      const lnurlLink = target.closest(
+        '[href^="lnurlp:" i],[href^="lnurlw:" i],[href^="lnurlc:" i]'
+      );
       const bitcoinLinkWithLighting = target.closest(
         '[href*="lightning=ln" i]'
-      ); // links with a lightning parameter and a value that starts with ln: payment requests (lnbc...) or lnurl (lnurl*)
+      ); // links with a lightning parameter and a value that starts with ln: payment requests (lnbc...) or lnurl (lnurl[pwc]:)
```
Comment (ekzyis) [inline src/extension/inpage-script/index.js:53]: There might be an error (unrelated to this PR) in this comment: Shouldn't it be `lightning:` and not `ln:`?

```diff
- // links with a lightning parameter and a value that starts with ln: payment requests (lnbc...) or lnurl (lnurl[pwc]:)
+ // links with a lightning parameter and a value that starts with lightning: payment requests (lnbc...) or lnurl (lnurl[pwc]:)

Comment (pavanjoshi914) [reply src/extension/inpage-script/index.js:53]: no changes are required there as it indicates links with lightning parameter + value that starts with ln. for eg. lnbc or lnurl

**Comment by pavanjoshi914** [comment]
had a round of tests. LGTM cc @rolznz  the error is just because we use more generalised matcher which can cause such error with relative links having lnurl inlcuded in the link

File: src/extension/inpage-script/index.js [hunk 2]
```diff
@@ -45,10 +45,12 @@ function registerLightningLinkClickHandler() {
         return;
       }
       const lightningLink = target.closest('[href^="lightning:" i]');
-      const lnurlLink = target.closest('[href^="lnurl" i]');
+      const lnurlLink = target.closest(
```
Comment (rolznz) [inline src/extension/inpage-script/index.js:48]: Could we link to the relevant spec? I think it is LUD-17

Comment (rolznz) [reply src/extension/inpage-script/index.js:48]: Should we support keyauth too?

Comment (ekzyis) [reply src/extension/inpage-script/index.js:50]: > Could we link to the relevant spec? I think it is LUD-17

Done in 385af36

> Should we support keyauth too? (I think this change will currently break it)

I think you never supported keyauth so this change will not break it. I searched for `keyauth` in your source code and wasn't found:

```
$ grep -R "keyauth" src/
```

Comment (rolznz) [reply src/extension/inpage-script/index.js:50]: Ah ok, thanks! I'll made a new issue for that: https://github.com/getAlby/lightning-browser-extension/issues/2984

**Review by rolznz — APPROVED** [review]
utACK

**Pull request merged**